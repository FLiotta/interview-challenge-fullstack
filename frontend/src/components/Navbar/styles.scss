@import '../../styles/colors.scss';

.navbar {
  width: 100%;
  padding: 25px;
  display: flex;
  justify-content: space-between;

  & > p, span, strong, small {
    margin-bottom: 0;
  }

  &__chip {
    display: flex;
    align-items: center;
    border-radius: 6px;
    padding: 8px 15px;
    cursor: pointer;
    transition: 0.5s all;

    &:hover {
      background-color: #eee;
    }

    &-username {
      font-family: 'montserrat';
      margin-bottom: 0px;
      margin-right: 15px;
    }
  }
}