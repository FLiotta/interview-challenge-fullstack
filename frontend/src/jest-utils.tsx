// @Packages
import { ReactNode } from 'react';
import { render as rtlRender } from '@testing-library/react';
import { Provider } from 'react-redux';

// @Project
import store from 'store';
import { JsxElement } from 'typescript';

function render(
  ui: ReactNode | JsxElement | any,
  renderOptions: any = { }
) {
  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>{children}</Provider>
    )
  }
  return rtlRender(ui, {
     wrapper: Wrapper, 
     ...renderOptions
    })
}

export * from '@testing-library/react'

export { render }
