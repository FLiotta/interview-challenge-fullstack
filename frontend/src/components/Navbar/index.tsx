// @Packages
import React from 'react';
import { useSelector } from 'react-redux';

// @Project
import Logo from 'components/Logo';
import ProfileBadge from 'components/ProfileBadge';

// @Own
import './styles.scss';

const Navbar: React.FC<any> = () => {
  

  return (
    <div className="navbar">
      <Logo brandColor tag="h6" />
      <div className="navbar__ctas">
        <ProfileBadge />
      </div>
    </div>
  )
}

export default Navbar;