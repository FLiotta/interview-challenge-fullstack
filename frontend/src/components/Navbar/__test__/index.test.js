// @Package
import { render } from 'jest-utils';
import { Provider } from 'react-redux';

// @Project
import Navbar from '../index';

describe('Navbar component', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <Navbar />
    );

    expect(container).toMatchSnapshot();
  })
})