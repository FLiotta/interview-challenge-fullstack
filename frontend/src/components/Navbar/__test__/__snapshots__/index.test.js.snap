// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component Should match snapshot 1`] = `
<div>
  <div
    class="navbar"
  >
    <h6
      class="logo logo--brand-color"
      data-testid="logo-testid"
    >
      PiggyBank
    </h6>
    <div
      class="navbar__ctas"
    >
      <div
        class="profilebadge"
        data-testid="profilebadge-testid"
      >
        <p
          class="profilebadge__username"
        >
           
        </p>
        <div
          class="avatar"
          data-testid="avatar-testid"
          style="width: 30px; height: 30px;"
        >
          <div
            class="avatar__fill avatar__fill--1"
            data-testid="avatar-fill-testid"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
